lib/pi_piper/bcm2835.rb

```diff
@@ -7,6 +7,8 @@ module Bcm2835
     extend FFI::Library
     ffi_lib File.dirname(__FILE__) + '/libbcm2835.img'
 
+    @@pins = []
```

Comment:
I don't think you need to use class variables (@@) here because it's a module. A regular instance variable (@) should work just fine (see: http://makandracards.com/makandra/14229-the-many-gotchas-of-ruby-class-variables)